core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/groupBy.kt

```diff
@@ -21,14 +36,289 @@ import kotlin.reflect.KProperty
+ * defines a group consisting of all rows where the column(s) contain that value combination.
  *
- * @param cols key columns; Column for grouping can be created inplace
+ * Returns a [GroupBy] — a dataframe-like structure that contains all unique combinations of key values
+ * along with the corresponding groups of rows (each represented as a [DataFrame]).
  *
- * `df.groupBy { expr("columnName") { "someColumn"<Int>() + 15 } }`
+ * A [GroupBy] can then be:
+ * * [transformed][Transformation] into a new [GroupBy];
+ * * [reduced][Reducing] into a [DataFrame], where each group is collapsed into a single representative row;
+ * * [aggregated][Aggregation] into a [DataFrame], where each group is transformed into one or more rows of derived values;
+ * * [pivoted][Pivoting] into a [PivotGroupBy] structure, which combines [pivot] and [groupBy] operations.
  *
- * is equivalent to
+ * Grouping keys can also be created inline:
+ * ```kotlin
+ * // Create a new column "newName" based on existing "oldName" values
+ * // and use it as a grouping key:
+ * df.groupBy { expr("newName") { oldName.drop(5) } }
+ * ```
  *
- * `df.add("columnName") { "someColumn"<Int>() + 15 }.groupBy("columnName")`
+ * Check out [Grammar].
+ *
+ * @include [SelectingColumns.ColumnGroupsAndNestedColumnsMention]
+ *
+ * See [Selecting Columns][GroupBySelectingOptions].
+ *
+ * For more information: {@include [DocumentationUrls.GroupBy]}
+ *
+ * Don't confuse this with [group], which groups column into
+ * [column group][org.jetbrains.kotlinx.dataframe.columns.ColumnGroup].
+ */
+internal interface GroupByDocs {
+    /**
+     * ## [groupBy][groupBy] Operation Grammar
+     * {@include [LineBreak]}
+     * {@include [DslGrammarLink]}
+     * {@include [LineBreak]}
+     *
+     * ### Create and transform [GroupBy]
+     *
+     * [**`groupBy`**][groupBy]**`(`**`moveToTop: `[`Boolean`][Boolean]**` = true)  {  `**`columns: `[`ColumnsSelector`][ColumnsSelector]**` }`**
+     *
+     * {@include [Indent]}
+     * `\[ `__`.`__[**`sortByGroup`**][GroupBy.sortByGroup]**`() `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByGroupDesc`**][GroupBy.sortByGroupDesc]**`() `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByCount`**][GroupBy.sortByCount]**`() `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByCountAsc`**][GroupBy.sortByCountAsc]**`() `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByKey`**][GroupBy.sortByKey]**`() `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByKeyDesc`**][GroupBy.sortByKeyDesc]**`() `**`]`
+     *
+     * {@include [Indent]}
+     * `\[ `__`.`__[**`sortBy`**][GroupBy.sortBy]**`  {  `**`columns: `[`ColumnsSelector`][ColumnsSelector]**` } `**
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`sortByDesc`**][GroupBy.sortByDesc]**`  {  `**`columns: `[`ColumnsSelector`][ColumnsSelector]**` } `**`]`
```

Comment:
same story with the 'or'. I'd just write `[]`, it's hard to tell the difference between `[` and `|`, plus `[]` is more accurate